src/domain-functions.ts

```diff
@@ -30,23 +37,10 @@ function all<Fns extends DomainFunction[]>(
   ...fns: Fns
 ): DomainFunction<UnpackAll<Fns>> {
   return ((input, environment) => {
-    return safeResult(async () => {
-      const results = await Promise.all(
-        fns.map((fn) => (fn as DomainFunction)(input, environment)),
-      )
-
-      if (!isListOfSuccess(results)) {
-        throw new ResultError({
-          errors: results.map(({ errors }) => errors).flat(),
-          inputErrors: results.map(({ inputErrors }) => inputErrors).flat(),
-          environmentErrors: results
-            .map(({ environmentErrors }) => environmentErrors)
-            .flat(),
-        })
-      }
-
-      return results.map(({ data }) => data)
-    })
+    const [first, ...rest] = fns.map((df) =>
+      atmp(() => fromSuccess(df)(input, environment)),
+    )
+    return dfResultFromAtmp(A.all(first, ...rest))()
```

Comment:
Why do we need the destructuring here?
Wouldn't this work?
```suggestion
    const results = fns.map((df) =>
      atmp(() => fromSuccess(df)(input, environment)),
    )
    return dfResultFromAtmp(A.all(results))()
```

Reply:
It does not type check. I just assumed that the type was like that by design, but I could be wrong.